packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -118,14 +141,28 @@ export class OverflowList<T> extends React.PureComponent<IOverflowListProps<T>,
         ) {
             // reset visible state if the above props change.
             this.setState({
+                direction: OverflowDirection.GROW,
+                lastOverflow: undefined,
                 overflow: [],
                 visible: nextProps.items,
             });
         }
     }
 
-    public componentDidUpdate() {
+    public componentDidUpdate(_prevProps: IOverflowListProps<T>, prevState: IOverflowListState<T>) {
         this.repartition(false);
```

Comment:
nit: I'd destructure here to avoid repeated lookups into `this.state` fields. I don't think you'd lose any clarity.
```tsx
const { direction, lastOverflow, overflow } = this.state;
```

Reply:
done below